### PR TITLE
Cc feature/16/subscribed posts on homepage

### LINKED ISSUE #16: View Subscribed Posts on Homepage
As a reader I would like to see a list of Posts written by authors I am subscribed to, so that I can more easily find Posts I may be interested in.

**Given** A user is on the homepage  
**When** the user has at least one Subscription  
**Then** they should see a list of Posts written by authors they are subscribed to

### PR DESCRIPTION
**Pull request by ccote95**
### Purpose
To implement the home page being a list of posts by users the loggedIn user is subscribed to

### Files Changed
- Created the homePage component and rendered posts by users the loggedin user is subscribed to in `HomePagePostList.jsx`
- Added the element to the homepage route in `ApplicationViews.jsx`
- Created a controller endpoint to handle getting the subscribed to posts in `PostController.cs`
- Created a fetch to get the subscribed to posts in `postManager.js`

### Testing
Fetch, setup, and run according to the project README. but do not create a new migration
Confirm the following:
- When logged in as a user thats subscribed to another user, a list of posts created by that user should be rendered

closes #16

### CODE REVIEW

File: client/src/components/posts/HomePagePostList.jsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        <div className="w-75 d-flex align-items-center justify-content-between" style={{maxWidth: "1200px"}}>
+            <h1>Subscribed Posts</h1>
+            {/* {loggedInUser.roles.includes("Admin") && (
+                <Button 
+                    color = "primary" 
+                    onClick={() => {navigate("unapproved")}}
+                >
+                    Unapproved Posts: {unapprovedCount}
+                </Button>
+            )} */}
```
Comment (Michael-Deitz) [inline client/src/components/posts/HomePagePostList.jsx:24]: Line 17 - 24 is there a reason to leave the commented out code snippet?

File: Controllers/PostController.cs [hunk 2]
```diff
@@ -300,4 +300,51 @@ public IActionResult Approve(int id)
 
     }
 
+    [HttpGet("{id}/subscribedPosts")]
+    public IActionResult GetSubscribedPosts(int id)
+    {
+        List<Subscription> subs = _dbContext.Subscriptions.Where(s => s.FollowerId == id).ToList();
+
+        var commonAttribute = subs.Select(s => s.CreatorId).ToList();
```
Comment (ztrouy) [inline Controllers/PostController.cs:308]: Change from var to a list of integers, and change the variable name from `commonAttribute` to something more descriptive

File: client/src/components/posts/HomePagePostList.jsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import { Badge, Button, Card, CardLink, CardText} from "reactstrap"
+
+export default function HomePagePosts({loggedInUser})
+{
+    const [subscribedPosts, setSubscribedPosts] = useState([])
+
+    useEffect(() => {
+        getSubscribedPosts(loggedInUser.id).then(setSubscribedPosts)
+    },[])
+    return (
```
Comment (ztrouy) [inline client/src/components/posts/HomePagePostList.jsx:13]: Could you add a line above this return to give it some space

File: client/src/managers/postManager.js [hunk 4]
```diff
@@ -53,4 +53,8 @@ export const approvePost = (postId) => {
 
 export const getUnapprovedCount = () => {
     return fetch(`${_apiUrl}/unapprovedCount`).then(res => res.json());
+}
+
+export const getSubscribedPosts = (Id) => {
```
Comment (ztrouy) [inline client/src/managers/postManager.js:58]: should be "id", not "Id"

**Review by ztrouy — CHANGES_REQUESTED** [review]

**Review by Michael-Deitz — APPROVED** [review]

**Review by ztrouy — APPROVED** [review]
:shipit:

**Pull request merged**